app/api/v0/proofs/download/[id]/route.ts

```diff
@@ -10,11 +10,17 @@ export async function GET(
 
   const { data, error } = await client
     .from("proofs")
-    .select("proof, block_number, cluster_id, user_id")
+    .select("block_number, cluster_id, user_id")
     .eq("proof_id", id)
     .single()
 
-  if (error || !data.proof) {
+  const { data: binaryData, error: binaryError } = await client
+    .from("proof_binaries")
+    .select("proof_binary")
+    .eq("proof_id", id)
```

Comment:
We could add this query with a join in the previous one to optimize the requests.